### PR TITLE
fix: derive running ws stop time from deadline

### PR DESCRIPTION
**Pull request by greyscaled**
Summary:

When a workspace is on, the remaining time until shutdown needs to be
derived from the deadline timestamp, not implied from the TTL. A utility is
function is added that answers the question if a workspace is on.

Impact:

This is a shared piece of logic in workspace scheduling presentations.
In particular it unblocks work in 1779, or at least allows an
implementation that shares details with the WorkspaceScheduleBanner.

Notes:

We could possibly instead return whether the workspace is "ON",
"UNKNOWN", or "OFF". Maybe a future improvement for that could be made
as the neds arrises.

### CODE REVIEW

File: site/src/util/workspace.test.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+      ["stop", "pending", false],
+      ["stop", "running", false],
+      ["stop", "succeeded", false],
+
+      ["start", "canceled", false],
+      ["start", "canceling", false],
+      ["start", "failed", false],
+      ["start", "pending", false],
+      ["start", "running", false],
+      ["start", "succeeded", true],
```
Comment (johnstcn) [inline site/src/util/workspace.test.ts:27]: 💝

File: site/src/util/workspace.ts [hunk 2]
```diff
@@ -185,3 +185,9 @@ export const getDisplayAgentStatus = (
       }
   }
 }
+
+export const isWorkspaceOn = (workspace: Workspace): boolean => {
+  const transition = workspace.latest_build.transition
+  const status = workspace.latest_build.job.status
+  return transition === "start" && status === "succeeded"
+}
```
Comment (johnstcn) [inline site/src/util/workspace.ts:193]: 👍  it's a narrow definition of 'on' but I think that's the correct one.

**Review by johnstcn — APPROVED** [review]
BE ✔️

**Comment by vapurrmaid** [comment]
This resolved #1779

**Pull request merged**